### DISCUSSION TITLE
Rename a GitHub Action on the Marketplace

### DISCUSSION THREAD

**Discussion by wouter-admiraal-sonarsource** [community/community#27118] [Actions]
<p>Hi all,</p>
<p>We’re about to publish a GitHub Action to the Marketplace, but are unsure about the (human readable) name. Once we publish our action to the Marketplace, can we still change the name in the <code>action.yml</code> file, and will this be reflected in the Marketplace? Or is the name set in stone the first time it’s published? Note that we will <em>not</em> change the repo slug, which will remain the same regardless of our naming strategy. I’m only speaking about the <code>name</code> property of the <code>action.yml</code> file.</p>
<p>Thanks for your help!</p>

**Comment by jsoref** [comment, marked as answer]
<p>Changing the name in the yaml will change the url in the marketplace:</p><aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/desbo/merge-pr-action/issues/10#issuecomment-844994834" target="_blank" rel="noopener nofollow ugc">github.com/desbo/merge-pr-action</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/desbo/merge-pr-action/issues/10#issuecomment-844994834" target="_blank" rel="noopener nofollow ugc">Action name misspells `dependency`</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-05-20" data-time="09:35:20" data-timezone="UTC">09:35AM - 20 May 21 UTC</span>
      </div>

        <div class="date">
          closed <span class="discourse-local-date" data-format="ll" data-date="2021-05-20" data-time="15:28:11" data-timezone="UTC">03:28PM - 20 May 21 UTC</span>
        </div>

      <div class="user">
        <a href="https://github.com/jsoref" target="_blank" rel="noopener nofollow ugc">
          <img alt="jsoref" src="https://user-images.githubusercontent.com/2119212/181101775-df7f8c33-2a3f-4deb-bd6d-178be1b3f0cc.png" class="onebox-avatar-inline" width="20" height="20">
          jsoref
        </a>
      </div>
    </div>

    <div class="labels">
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">https://github.com/desbo/merge-pr-action/blob/e5cd901e774423f1087298b4c3479b659b<span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden">d1f0f9/action.yml#L1</span></p>
  </div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Consumers are tied to the org+repository name and not the pretty name, so, it’s mostly not a huge deal, unless you have a lot of people linking to the marketplace url.</p>
<p>Do note that org/org or user/user is now a magical repository name, so while that may have been the logical choice for an action, it’s now not a good idea. Thus, org/org-action may be a better choice.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <a href="https://torrocus.com/blog/special-github-repository/" target="_blank" rel="noopener nofollow ugc" title="09:55PM - 11 July 2020">Less is more – 11 Jul 20</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181101776-bb88aecd-3ae7-4e4a-aa4f-5373f29dfbd4.png" class="thumbnail onebox-avatar" width="144" height="144">

<h3><a href="https://torrocus.com/blog/special-github-repository/" target="_blank" rel="noopener nofollow ugc">Create a special repository in your GitHub Profile 🔨</a></h3>

<p>How to add README.md to GitHub Profile? What features are supported and which are not?</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Also note that renaming your default branch will break consumers, so if you plan to switch from <code>master</code> to <code>main</code>, it’s a good idea to do so <em>before</em> you advertise. GitHub’s magical rename feature does <em>not</em> support this.</p>

**Comment by wouter-admiraal-sonarsource** [comment]
<p>Thanks for the answer, and the link :+1:</p>
